filebeat/beat/spooler.go

```diff
@@ -112,6 +113,12 @@ func (s *Spooler) Stop() {
 
 // flush flushes all event and sends them to the publisher
 func (s *Spooler) flush() {
+
+	// Prevents race in flush method
+	var mutex = &sync.Mutex{}
```

Comment:
I will check this PR's AppVeyor test output for Filebeat to confirm that the data race is fixed. Just have to wait :watch: for AppVeyor to finished.